test/dx_e2e.js

```diff
@@ -1,56 +1,115 @@
+    await dx.claimSellerFunds(secondaryTokenMock.address, depositToken.address, accounts[0], 1)
+    await dx.claimBuyerFunds(depositToken.address, secondaryTokenMock.address, accounts[0], 1)
+    await dx.claimBuyerFunds(secondaryTokenMock.address, depositToken.address, accounts[0], 1)
+    await dx.postSellOrder(depositToken.address, secondaryTokenMock.address, 0, oneGwei.toString())
+    await dx.postSellOrder(secondaryTokenMock.address, depositToken.address, 0, oneGwei.toString())
+
+    //close the auction and pool paritcpation in next one
+    await waitUntilPriceIsXPercentOfPreviousPrice(depositToken, secondaryTokenMock, 1);
+    await dx.postBuyOrder(depositToken.address, secondaryTokenMock.address, 2, oneGwei.toString())
+    await dx.postBuyOrder(secondaryTokenMock.address, depositToken.address, 2, oneGwei.toString())
+
+    assert.equal(await dx.getAuctionIndex(depositToken.address, secondaryTokenMock.address),3)
+    console.log("got second auction finished")
+    await coordinator.participateInAuction()
+
+    //claim funds  and post into next auction auction with more
+    await dx.claimSellerFunds(depositToken.address, secondaryTokenMock.address, accounts[0], 2)
+    await dx.claimSellerFunds(secondaryTokenMock.address, depositToken.address, accounts[0], 2)
+    await dx.claimBuyerFunds(depositToken.address, secondaryTokenMock.address, accounts[0], 2)
+    await dx.claimBuyerFunds(secondaryTokenMock.address, depositToken.address, accounts[0], 2)
+    await dx.postSellOrder(depositToken.address, secondaryTokenMock.address, 0, oneGwei.toString())
+    await dx.postSellOrder(secondaryTokenMock.address, depositToken.address, 0, oneGwei.toString())
+
+    //close the auction and pool paritcpation in next one
+    await waitUntilPriceIsXPercentOfPreviousPrice(depositToken, secondaryTokenMock, 0.9999);
+    await dx.postBuyOrder(depositToken.address, secondaryTokenMock.address, 3, oneGwei.toString())
+    await dx.postBuyOrder(secondaryTokenMock.address, depositToken.address, 3, oneGwei.toString())
+
+    assert.equal(await dx.getAuctionIndex(depositToken.address, secondaryTokenMock.address),4)
+    console.log("got third auction finished")
+
+    // end pool trading period:
+    await waitForNBlocks(100, accounts[0])
 
-    //  await depositTokenMock.givenAnyReturnUint(10)
-    //  const balanceOf = erc20.contract.methods.balanceOf(instance.address).encodeABI()
-    //  await depositTokenMock.givenMethodReturnUint(balanceOf, 10)
-    //  await instance.participateInAuction()
+    await instance1.triggerMGNunlockAndClaimTokens()
+    
+    console.log("could withdraw balance")
+    const balBefore = await depositToken.balanceOf.call(accounts[0]);
+    await instance1.withdrawDeposit()
+    const balAfter = await depositToken.balanceOf.call(accounts[0]);
 
-  	  // assert.equal(await dx.sellerBalance.call(DxMgnPool.address, 0), 10)*/
+    assert.equal(balAfter.sub(balBefore).toString(), 9);
```

Comment:
Why exactly 9 difference?

Reply:
We should get 10 out of it, but the second auction the price is not 1, it's 0.999999 and then it rounds down from 10 to 9.